.github/workflows/test.yml

```diff
@@ -0,0 +1,119 @@
+name: "Build and Test virtual routers"
+
+on: [push, pull_request]
```

Comment:
Oh, I thought it would detect a workflow already exists for *push* and not create a new run for a pull request for the same commit?!

Reply:
And then you push new commits to an existing PR and again it runs twice. Maybe we should change it to pull-request and push-to-main?

Reply:
I do PR & push to main for the Acton repo but there is an alternative way.

Using Github actions concurrency bla bla, you can limit the concurrency for a given key and cancel one pipeline, see https://docs.github.com/en/enterprise-cloud@latest/actions/using-jobs/using-concurrency

Reply:
Done.